topics/threaded-io.md

```diff
@@ -0,0 +1,127 @@
+# Threaded I/O
+## Introduction
```

Comment:
Should we mention why is threading avoided in Redis ( simplicity among others .... )

Reply:
Added a few desc about this:
- simple and reliable
- entry barrier for every new feature